src/Connections/GenericHook.ts

```diff
@@ -197,22 +196,29 @@ export class GenericHookConnection extends BaseConnection implements IConnection
         GenericHookConnection.LegacyCanonicalEventType,
     ];
 
-    private transformationFunction?: Script;
+    private state: GenericHookConnectionState;
+    private hasValidTransformation = false;
     private cachedDisplayname?: string;
     constructor(
         roomId: string,
-        private state: GenericHookConnectionState,
-        public readonly hookId: string,
-        stateKey: string,
+        stateContent: Record<string, unknown>,
+        hookId: string,
         private readonly messageClient: MessageSenderClient,
         private readonly config: BridgeConfigGenericWebhooks,
         private readonly as: Appservice,
         private readonly intent: Intent,
     ) {
-        super(roomId, stateKey, GenericHookConnection.CanonicalEventType);
-        if (state.transformationFunction && config.allowJsTransformationFunctions) {
-            this.transformationFunction = new Script(state.transformationFunction);
-        }
+        const validState = GenericHookConnection.validateState(stateContent);
+        super(roomId, validState.name, GenericHookConnection.CanonicalEventType);
+        this.state = {
+            ...validState,
+            hookId,
+        };
+        void this.validateTransformationFunction();
```

Comment:
The previous version didn't actually validate the script at this point (I believe new Script() wouldn't do a thing until it was run). As much as it might feel weird, I'd rather not break that behaviour for now.

Sending matrix messages in the constructor is a *big* no-no too.

Reply:
Ok, I'll revert the validation in the constructor, but will at least add a docstring to say that the `state` argument ought to come from a call to `validateState`.

Reply:
Validation reverted and docstring added.